urlimport.go

```diff
@@ -0,0 +1,212 @@
+package main
+
+import "fmt"
+import "flag"
+import "net/http"
+import "os"
+import "io"
+import "io/ioutil"
+import "net/url"
+import "strings"
+import "time"
+import "net"
+
+import "github.com/jlaffaye/ftp"
+
+var filename string
+var downloadURL string
+var log *os.File
+var logfileName string
+
+const (
+	defaultPort = "21"
+	defaultUser = "anonymous"
+	defaultPass = "anonymous"
+	logfileExtn = "-url-import-error.txt"
+)
+
+func main() {
+	logfileName = strings.Join(strings.Split(time.Now().Format(time.UnixDate), " "), "-") + logfileExtn
+	file, err := os.Create(logfileName)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "Unable to create error file. ", err)
+		os.Exit(1)
+		return
+	}
+	log = file
+	os.Stderr = file
+	defer file.Close()
+
+	parseArgs()
+	scheme, host, port, user, pass, path, exitcode := ParseURL()
+	if exitcode != 0 {
+		cleanup()
+		defer os.Exit(exitcode)
+		return
+	}
+
+	switch strings.ToLower(scheme) {
+	case "http", "https":
+		exitcode = DownloadFromURL()
+	case "ftp", "sftp":
+		exitcode = DownloadFromFtp(host, port, user, pass, path)
+	default:
+		logMessage("Invalid Scheme.")
+	}
+
+	cleanup()
+	defer os.Exit(exitcode)
+}
+
+//cleanup clean up o byte log file or partially downloaded output file
+func cleanup() {
+	// if error file is 0 bytes, then remove it
+	if fi, err := os.Stat(logfileName); !os.IsNotExist(err) {
+		if fi.Size() == 0 { // no errors logged
+			os.Remove(logfileName)
+		} else {
+			os.Remove(filename) //there were some errors logged. Remove partial downloaded file.
+		}
+	}
+}
+
+//logMessage to a log file
+func logMessage(message string) {
+	log.WriteString(message)
+}
+
+//parseArgs parse command line args
+func parseArgs() int {
+	flag.StringVar(&filename, "filename", "", "file name to use when saving the imported file")
+	flag.StringVar(&downloadURL, "url", "", "Url to import the file from")
+	flag.Parse()
```

Comment:
This is possibly unneeded, but I like the way the `flag` library is used in e.g. `de-job-killer`. In that construction, you specify the flag-related stuff when declaring the variable. I think it returns a pointer, which would change how some things are called, but probably not in too annoying a way.